resolver.go

```diff
@@ -233,15 +235,27 @@ func (r *resolver) ServeDNS(w dns.ResponseWriter, query *dns.Msg) {
 		for i := 0; i < num; i++ {
 			log.Debugf("Querying ext dns %s:%s for %s[%d]", w.LocalAddr().Network(), r.extDNS[i], name, query.Question[0].Qtype)
 
-			c := &dns.Client{Net: w.LocalAddr().Network()}
-			addr := fmt.Sprintf("%s:%d", r.extDNS[i], 53)
+			var extConn net.Conn
+			r.sb.execFunc(func() {
```

Comment:
Let's not get into the container namespace every time in the critical path when we need to make an external query. Let's create these connections during sandbox creation time and then just it here when we need to do external query.